### PR TITLE
do not reuse http requests

### PR DESCRIPTION
**Pull request by bolshoy**
reusing http.Request objects is problematic due to contention on some
internal fields (Header, etc). Instead, a new request object is
created on each http request.

### CODE REVIEW

**Comment by yosriz** [comment]
@bolshoy can you assign someone from your team to review this?

File: rpc/http.go [hunk 1]
```diff
@@ -65,21 +65,14 @@ func (hc *httpConn) Close() error {
 
 // DialHTTP creates a new RPC clients that connection to an RPC server over HTTP.
 func DialHTTP(endpoint string) (*Client, error) {
```
Comment (oryband) [inline rpc/http.go:67]: is verifying the endpoint is a legal string necessary? just a thought

File: rpc/http.go [hunk 2]
```diff
@@ -65,21 +65,14 @@ func (hc *httpConn) Close() error {
...
-	if err != nil {
-		return nil, err
-	}
-	req.Header.Set("Content-Type", "application/json")
-	req.Header.Set("Accept", "application/json")
-
 	initctx := context.Background()
 	return newClient(initctx, func(context.Context) (net.Conn, error) {
 		// add cookie support
 		cookieJar, _ := cookiejar.New(nil)
```
Comment (oryband) [inline rpc/http.go:71]: why ignoring the error?

**Review by oryband — APPROVED** [review]
looks good, asked some questions.

File: rpc/http.go (hunk 2 above)
Comment (bolshoy) [reply rpc/http.go:71]: nothing bad except out of memory can happen here

File: rpc/http.go (hunk 1 above)
Comment (bolshoy) [reply rpc/http.go:67]: maybe it's already done in the caller, have you checked?

**Pull request merged**